rosdep/python.yaml

```diff
@@ -8731,8 +8731,10 @@ python3-ruamel.yaml:
     buster: [python3-ruamel.yaml]
     jessie: [python3-ruamel.yaml]
     stretch: [python3-ruamel.yaml]
+  fedora: [python3-ruamel-yaml]
   nixos: [python3Packages.ruamel_yaml]
   openembedded: [python3-ruamel-yaml@meta-python]
+  rhel: ['python%{python3_pkgversion}-ruamel-yaml']
```

Comment:
@cottsay hmm, either `%{python3_pkgversion}` isn't necessary here or it's necessary in #34817.

Reply:
There was a period earlier in the RHEL 7 cycle where Python 3 packages shipped by EPEL were packaged with a `python36-` prefix but did not provide a `python3-` virtual package, which should have started happening when Python 3.6 became the default Python 3 version for RHEL 7.

Unfortunately, some Python 3.6 packages either haven't been rebuilt since then, or fail to include the necessary macros to provide the `python3-` virtual package. In those cases, we need to use `%{python3_pkgversion}`, which resolves to `36` on RHEL 7 and `3` pretty much everywhere else.

Using the macro is confusing to say the least, but it is only necessary on some older packages. As an alternative, we could split the rule and have separate rules for RHEL 7 and "other" RHEL (8 and 9). When this pattern was introduced, it was believed that we'd be dealing with a changing `%{python3_pkgversion}` value on a regular basis, which turned out to be untrue, so it's now only used in special circumstances.

Many current uses of `%{python3_pkgversion}` throughout this file could probably be switched to just using a `python3-` prefix instead.

So that's the story, and this package is one of the few legacy-ish packages that don't provide the `python3-` virtual package like they should.

Reply:
Thanks for the thorough explanation @cottsay !